.changeset/angry-plums-retire.md

```diff
@@ -0,0 +1,7 @@
+---
+"@comet/cms-api": minor
```

Comment:
I believe this is a breaking change, as there are already projects which use Comet v6 and the users permissions module. To prevent this being breaking, we'd need to:

1. Keep the `Permission` interface in public API of `@comet/cms-api`
2. Deprecate `availablePermissions` in the module config instead of removing it

Also, I'd like to discuss keeping the interface for developer experience: This would prevent accidental typos, and also, how should devs know about available permissions in the library? For instance, if someone wanted to add a resolver that relies on the page tree?

Reply:
true. Could and should be made compatible.

> This would prevent accidental typos

So you try to avoid typos by letting the dev define it in 3 places instead of 1. Imho that is unnecessary extra work with no  improvement.

> how should devs know about available permissions in the library?

add them to docs? Or export as constant object?

> For instance, if someone wanted to add a resolver that relies on the page tree?

wouldn't you add almost always add a new permission?

Reply:
> This was my wish too, but I didn't find a way to accomplish this. A type like keyof Permission | string (and we need the string here) is widened by the IDE just to string and doesn't suggest the keys.

Haven't thought about the string widening. Could we remove the `| string` by making some `as keyof Permission` casts?

Reply:
> So you try to avoid typos by letting the dev define it in 3 places instead of 1. Imho that is unnecessary extra work with no improvement.

It would be in two places: 1. the interface module augmentation 2. the decorator. I'd still remove availablePermissions. Also, removing a permission from the interface would cause type errors in the decorators, further preventing accidentally forgetting a removed permission. 

However, do we even plan to support removing permissions?

Reply:
> it would be in two places:

ok, an improvement at least. Still I'd prefer 1 place.

The initial idea was from a api-generator user. The api-generator auto-generates the permission from the entity name, the workflow create-entity-and-run-generator somehow needs this extra permnission step if you have the interface.